concent_api/gatekeeper/tests/test_gatekeeper_views.py

```diff
@@ -0,0 +1,141 @@
+        encrypted_token = b64encode(golem_upload_token).decode()
+        response = self.client.post('/gatekeeper/upload-auth/blender/benchmark/test_task/scene-Helicopter-27-cycles.blend',
+            HTTP_AUTHORIZATION = 'Golem ' + encrypted_token,
+            content_type = 'application/x-www-form-urlencoded',
+            HTTP_CONCENT_CLIENT_PUBLIC_KEY = self.public_key
+        )
+
+        self.assertIsInstance(response, JsonResponse)
+        self.assertEqual(response.status_code, 401)
+        self.assertIn('message', response.json().keys())
+
+
+class GatekeeperViewDownloadTest(TestCase):
+
+    @freeze_time("2017-11-17 10:00:00")
+    def setUp(self):
+        self.client = Client()
+        self.message_timestamp = int(datetime.datetime.now().timestamp())  # 1510912800
+        self.public_key = '85cZzVjahnRpUBwm0zlNnqTdYom1LF1P1WNShLg17cmhN2UssnPrCjHKTi5susO3wrr/q07eswumbL82b4HgOw=='
+        self.upload_token = FileTransferToken(
+            token_expiration_deadline       = self.message_timestamp + 3600,
+            storage_cluster_address         = 'http://storage.concent.golem.network/',
+            authorized_client_public_key    = b'\xf3\x97\x19\xcdX\xda\x86tiP\x1c&\xd39M\x9e\xa4\xddb\x89\xb5,]O\xd5cR\x84\xb85\xed\xc9\xa17e,\xb2s\xeb\n1\xcaN.l\xba\xc3\xb7\xc2\xba\xff\xabN\xde\xb3\x0b\xa6l\xbf6o\x81\xe0;',
+            files                           = ['blender/benchmark/test_task/scene-Helicopter-27-cycles.blend', 'blender/benchmark/test_task/texture.tiff'],
+            operation                       = 'download'
+        )
+
+    @freeze_time("2017-11-17 10:00:00")
+    def test_download_should_accept_valid_message(self):
+        golem_upload_token = dump(self.upload_token, settings.CONCENT_PRIVATE_KEY, settings.CONCENT_PUBLIC_KEY)
+        encrypted_token = b64encode(golem_upload_token).decode()
+        response = self.client.get('/gatekeeper/download-auth/blender/benchmark/test_task/scene-Helicopter-27-cycles.blend',
+            HTTP_AUTHORIZATION = 'Golem ' + encrypted_token,
+            HTTP_CONCENT_CLIENT_PUBLIC_KEY = self.public_key
+        )
+
+        self.assertIsInstance(response, HttpResponse)
+        self.assertEqual(response.status_code, 200)
+
+    @freeze_time("2017-11-17 10:00:00")
+    def test_download_should_return_401_if_wrong_authorization_header(self):
+        golem_upload_token = dump(self.upload_token, settings.CONCENT_PRIVATE_KEY, settings.CONCENT_PUBLIC_KEY)
+        encrypted_token = b64encode(golem_upload_token).decode()
+        response1 = self.client.get('/gatekeeper/download-auth/blender/benchmark/test_task/scene-Helicopter-27-cycles.blend',
+            HTTP_AUTHORIZATION = 'GolemGolem ' + encrypted_token,
+            HTTP_CONCENT_CLIENT_PUBLIC_KEY = self.public_key
+        )
+
+        self.assertIsInstance(response1, JsonResponse)
+        self.assertEqual(response1.status_code, 401)
+        self.assertIn('message', response1.json().keys())
```

Comment:
You could just do it three times in a loop and put all the values of `HTTP_AUTHORIZATION` your're testing on a list. It would also make it easier for you to test more cases. For example you should also test: empty header, header consisting of just whitespace, header with more than one space, etc.

Reply:
I've made a loop with `wrong_test_headers = ['GolemGolem '+ encrypted_token, '', 'Golem encrypted_token ', 'Golem '+ 'a' + encrypted_token]`, but I left response2 to check if HTTP_AUTHORIZATION header exists at all.

Reply:
Oh, so you added additional `N` at the end of the header name to test with an invalid header. You should use someting more obvious. Like `HTTP_AUTHORIZATION_ABC`. I did not even notice that.

You can make the loop cover this too by defining a list of dicts instead of a list of strings:

``` python
wrong_test_headers = [
    {'HTTP_AUTHORIZATION':  'GolemGolem ' + encrypted_token},
    {'HTTP_AUTHORIZATIONN': 'GolemGolem ' + encrypted_token},
    {'HTTP_AUTHORIZATION':  ''},
    {'HTTP_AUTHORIZATION':  'Golem encrypted_token '},
    {'HTTP_AUTHORIZATION':  'Golem a' + encrypted_token},
]
```

Then pass the headers just like `kwargs`:
``` python
for headers in wrong_test_headers:
    response = Client().get(..., **headers)
    ...
```